internal/services/v1/relationships.go

```diff
@@ -126,30 +130,49 @@ func (ps *permissionServer) ReadRelationships(req *v1.ReadRelationshipsRequest,
 		return rewriteError(ctx, err)
 	}
 
-	usagemetrics.SetInContext(ctx, &dispatchv1.ResponseMeta{
-		DispatchCount: 1,
-	})
+	batchCount := datastoreQueryBatchSize
+	var cursor options.Cursor
+	var dispatchCount uint32
+	for batchCount == datastoreQueryBatchSize {
+		dispatchCount++
+		usagemetrics.SetInContext(ctx, &dispatchv1.ResponseMeta{
+			DispatchCount: dispatchCount,
+		})
+		tupleIterator, err := ds.QueryRelationships(
+			ctx,
+			datastore.RelationshipsFilterFromPublicFilter(req.RelationshipFilter),
+			options.WithLimit(&datastoreQueryBatchSize),
+			options.WithAfter(cursor),
+			options.WithSort(options.ByResource),
+		)
+		if err != nil {
+			return rewriteError(ctx, err)
+		}
 
-	tupleIterator, err := ds.QueryRelationships(ctx, datastore.RelationshipsFilterFromPublicFilter(req.RelationshipFilter))
-	if err != nil {
-		return rewriteError(ctx, err)
-	}
-	defer tupleIterator.Close()
+		batchCount = 0
+		for tpl := tupleIterator.Next(); tpl != nil; tpl = tupleIterator.Next() {
+			batchCount++
+			err := resp.Send(&v1.ReadRelationshipsResponse{
+				ReadAt:       revisionReadAt,
+				Relationship: tuple.ToRelationship(tpl),
+			})
+			if err != nil {
+				tupleIterator.Close()
+				return err
```

Comment:
we need to call `rewriteError` here as well or else de gRPC error will be returned with a different error code